### DISCUSSION TITLE
Welcome to edumips64 Discussions!

### DISCUSSION THREAD

**Discussion by lupino3** [EduMIPS64/edumips64#553] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

Feel free to use this post to introduce yourself, or just jump straight away into creating a new thread for your favourite topic. Thanks!

**Comment by shuckleFan** [comment]
Hi I'm a student that I have just started to use edumips64. I read the documentation, the part about system call, I don't get why regsiter R1 and register R14 are used for return value and parameter, respectively, instead of more conventional register. 
In particular register R1, isn't it reserved for assembler?
I hope this question is not too naive.

**Reply by lupino3** [reply]
Hi @shuckleFan and thanks for your question, which is definitely not naive :)

So, I did implement the `SYSCALL` instruction 17 years ago, but I do not remember at all why I chose those particular registers. As I look to online docs now, I think you are right and I should have used `r1` (`$at`) for the return value.

I suspect it wouldn't be a good idea to change behavior now, so I'll leave things as they are. I'll dig a bit further in my docs to see if I can find a reason for this choice :)

Thanks,
Andrea

**Reply by shuckleFan** [reply]
> I should have used r1 ($at) for the return value
Is that a typo?
You meant "should NOT" right?  🤔
 
by the way thank you for aswering me. i appreciate it a lot

**Reply by lupino3** [reply]
Yes, I meant *should not* have used `r1` :) apologies.
`$v0` (`$r2`) seems more appropriate.
